include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -350,4 +350,20 @@ def RegResetWithOneReset : Pat<
   (DropWrite $reg, (NodeOp $resetVal, $name, $nameKind, $annotations, $inner_sym)),
   [(OneConstantOp $reset)]>;
 
+//===----------------------------------------------------------------------===//
+// Subaccesses on InvalidValueOp
+//===----------------------------------------------------------------------===//
+
+// subfield(invalidvalue) -> invalidvalue
+def SubfieldOfInvalid : Pat<
+  (SubfieldOp (InvalidValueOp), $_), (InvalidValueOp)>;
+
+// subindex(invalidvalue) -> invalidvalue
```

Comment:
on these, can you make it clearer whether the invalid is the index or the aggregate?